lib/dal/src/attribute/value.rs

```diff
@@ -1944,4 +1946,72 @@ impl AttributeValue {
             None => None,
         })
     }
+
+    pub async fn get_parent_av_id_for_ordered_child(
+        ctx: &DalContext,
+        id: AttributeValueId,
+    ) -> AttributeValueResult<Option<AttributeValueId>> {
+        let workspace_snapshot = ctx.workspace_snapshot()?;
+
+        let ordering_node_id = match workspace_snapshot
+            .incoming_sources_for_edge_weight_kind(id, EdgeWeightKindDiscriminants::Ordinal)
+            .await?
+            .first()
+            .copied()
+        {
+            Some(ordering_idx) => workspace_snapshot.get_node_weight(ordering_idx).await?.id(),
+            None => return Ok(None),
+        };
```

Comment:
Probably `WorkspaceSnapshotGraph` or `WorkspaceSnapshot` should have this as its own method at some point.